### PR TITLE
Kettle general overview

### PR DESCRIPTION
**Pull request by MushuEE**
@amwat @spiffxp took a minute to summarize everything that happens in kettle for some Bus factor and general reference. (I think we should change it in the near future but till that time I wrote it as-is.

/area Kettle
/label tide/merge-method-squash

### CODE REVIEW

File: kettle/OVERVIEW.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Kettle is a service tasked with tracking and uploading to [BigQuery] all completed jobs with results stored in the list of [Buckets]. This data can be then used to track metrics on failures, flakes, activity, etc.
+
+This document is intended to walk maintainers through Kettle's workflow to get a better understanding of how to add features, fix bugs, or rearchitect the service. This document will be organized in a Top->Down way that will start at Kettle's ENTRYPOINT and work through each stage.
+
+## ENTRYPOINT
+Kettle's main process is the execution of `runner.sh` which:
+- sets gcloud auth credentials
+- creates initial "`bq config`"
+- pulls most recent [Buckets]
+- executes `update.py` on loof
```
Comment (amwat) [inline kettle/OVERVIEW.md:13]: s/loof/loop

File: kettle/OVERVIEW.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+- [stream.py](#Stream-Results): Wait for pub-sub events for completed builds and upload as results surface.
+
+# Make Database
+Flags:
+- --buckets (str): Path to YAML that defines all the gcs buckets to collect jobs from
+- --junit (bool): If true, collect Junit xml test results
+- --threads (int): Number of threads to run concurrently with
+- --buildlimit (int): **Used in staging*  colect only N builds on each job
+
+`make_db.py` does the work of determine all the builds to collect and store to the database. It aggrigates all the builds of two flavors: `pr` and `non-pr` builds. It searches gcs for build paths or generates build paths if they are "incremental builds" (monotomically increasing). It passes the work of collecting build information and results to threads that collect information. It then does a best-effort attempt to insert the build results to the DB, committing the instert every 200 builds.
```
Comment (amwat) [inline kettle/OVERVIEW.md:27]: s/aggrigates/aggregates

s/instert/insert

File: kettle/OVERVIEW.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+# Create JSON Results and Upload
+This stage gets run for each [BugQuery] table that Kettle is tasked with uploading data to. Typically looking like either:
```
Comment (amwat) [inline kettle/OVERVIEW.md:30]: s/BugQuery/BigQuery

**Review by amwat — COMMENTED** [review]
Thanks for summarizing this!

minor nits

File: kettle/OVERVIEW.md (hunk 3 above)
Comment (MushuEE) [reply kettle/OVERVIEW.md]: Thanks!

**Review by amwat — COMMENTED** [review]
/lgtm

**Pull request merged**